scripts/ci/reborn_changed_coverage.py

```
@@ -1,204 +1,424 @@
+            pending_start = index
+            depth = 0
+        if pending_start is None:
+            continue
+        excluded.add(index)
+        depth += line.count("{") - line.count("}")
+        if depth > 0:
+            continue
+        if "{" in line:
+            pending_start = None
+        elif index > pending_start and stripped and not stripped.startswith("#["):
+            pending_start = None
```

Ruff (0.16.0)

[warning] Combine `if` branches using logical `or` operator

Combine `if` branches

(SIM114)